sycl/source/detail/pi.cpp

```diff
@@ -79,6 +79,9 @@ bool bindPlugin(void *Library) {
 
   decltype(::piPluginInit) *PluginInitializeFunction = (decltype(
       &::piPluginInit))(getOsLibraryFuncAddress(Library, "piPluginInit"));
+  if (PluginInitializeFunction == nullptr)
+    return false;
```

Comment:
What do you think about returning some error code instead of just boolean from this function? By doing so, you can report better messages to the user (which might be helpful during debugging).

Right now I see at least two possible breakage points here:
- failed to get `piPluginInit` address, i.e. `PluginIintializeFunction == nullptr`
` PluginInitializeFunction` returned non-success error code, i.e. `err != PI_SUCCESS`

P.S. there is even `TODO` for this at line 92

Reply:
You are right. But this is needed at multiple places.
Whole PI functionality currently is not debugging friendly.
Is it ok if it is taken as a separate patch? @smaslov-intel and @romanovvlad have been suggesting a similar change.